core/src/main/scala/cats/mtl/MonadChronicle.scala

```diff
@@ -0,0 +1,29 @@
+package cats
+package mtl
+
+import cats.data.Ior
+
+trait MonadChronicle[F[_], E] extends Serializable {
+  val monad: Monad[F]
+
+  def dictate(c: E): F[Unit]
+
+  def disclose[A](c: E)(implicit M: Monoid[A]): F[A]
+
+  def confess[A](c: E): F[A]
+
+  def memento[A](ma: F[A]): F[Either[E, A]]
+
+  def absolve[A](a: A, ma: F[A]): F[A]
```

Comment:
Let's take that `A` by-name :)

With memento returning `Ior` this could be:
```scala
def absolve[A](default: => A, ma: F[A]): F[A] =
  memento(ma).map {
    case Left(_) => default
    case Right(a) => a
    case Both(_, a) => a
  }
```

Reply:
Since `memento` (or now `materialize`) returns an `F[E Ior A]`, we need to get under the `F`:

```scala
def absolve[A](a: => A, fa: F[A]): F[A] = monad.map(materialize(fa)) {
    case Ior.Left(_) => a
    case Ior.Right(a0) => a0
    case Ior.Both(_, a0) => a0
 }
```